### ISSUE TITLE
printer() causes Not a Constant Expression exception when using with Flutter, Dart2 preview

### ISSUE THREAD

**Issue by MichaelMarner** [brianegan/redux_logging#4]
Trying to initialise the store with logging middleware:

```dart
import 'package:redux_logging/redux_logging.dart';

class MyApp {
  final store = new Store<AppState>(appReducer,
    initialState: new AppState(), 
    middleware: [new LoggingMiddleware.printer()]
  );
}
```

But get the following exception at runtime:

    Exception has occurred.
    'package:redux_logging/redux_logging.dart': error: Not a constant expression.

Appears to be a Flutter/Dart 2 issue, works fine with Flutter and Dart 1.

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel beta, v0.1.5, on Mac OS X 10.13.3 17D102, locale en-AU)
[✓] Android toolchain - develop for Android devices (Android SDK 27.0.3)
[✓] iOS toolchain - develop for iOS devices (Xcode 9.2)
[✓] Android Studio (version 3.0)
[✓] VS Code (version 1.21.1)
[✓] Connected devices (1 available)
```

**Comment by brianegan** [comment]
Hey @MichaelMarner -- Thanks for the report, and sorry about the trouble!

Did you run into this problem using the Dart 2 version of this lib (0.2.x) or the Dart 1 version of this lib (0.1.x)? Thanks!

**Comment by brianegan** [comment]
Tracked down the problem. Thanks for reporting! A new version `0.3.0` has been deployed that fixes this bug and supports Redux 3.0.0

**Comment by MichaelMarner** [comment]
Thanks for fixing, and sorry I should have been more specific on the version I was using. But you were correct:

When testing with Dart 2 I was using 0.2.x
